src/effects/VST/VSTEffect.cpp

```diff
@@ -1344,15 +1344,18 @@ bool VSTEffect::LoadSettings(const CommandParameters& parms, EffectSettings& set
    long index{};
    wxString key;
    float value = 0.0f;
+   int intKey = 0;
    if (parms.GetFirstEntry(key, index))
    {
       do
       {
          if (parms.Read(key, value))
-            vstSettings.mParamsMap[key] = value;
+            vstSettings.mParamsMap[key] = { intKey, value };
          else
             return false;
 
+         intKey++;
```

Comment:
This doesn't look right.

What makes you sure this assignment of integer keys is correct?

Reply:
Actually it doesn't matter because those integers will never be used when part of EffectSettings, but only when part of a Message.

In fact, maybe the Message should not use a map at all, but instead a vector indexed by small integers?

Yes, in fact, maybe the analogy with AudioUnits is misleading.  Paramer IDs for a VST effect will be some small initial segment of integers of size `mAEffect->numParams` and you only need a vector of optional<float> in the Message.

It didn't work like this for AudioUnits because parameter IDs could be a sparse sequence of larger numbers, so a map made more sense there.